app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/agents/AgentAccessAuditService.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.services.agents
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.libs.json.{Json, Writes}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.agents.routes
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.address.{Country, Postcode}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.agents.audit.{AgentAccessAttempt, AgentVerifierMatchAttempt}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.ids.{AgentReferenceNumber, CgtReference}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions._
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
+
+import scala.concurrent.ExecutionContext
+@ImplementedBy(classOf[AgentAccessAuditServiceImpl])
+trait AgentAccessAuditService {
+
+  def sendAgentAccessAttemptEvent(
+    agentReferenceNumber: AgentReferenceNumber,
+    cgtReference: CgtReference,
+    success: Boolean
+  )(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Unit
+
+  def sendAgentVerifierMatchAttemptEvent(
+    agentReferenceNumber: AgentReferenceNumber,
+    cgtReference: CgtReference,
+    attemptsMade: Int,
+    maxAttempts: Int,
+    attemptedVerifier: Either[Country, Postcode],
+    success: Boolean
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Unit
+
+}
+
+@Singleton
+class AgentAccessAuditServiceImpl @Inject()(auditConnector: AuditConnector)
```

Comment:
Why not use the existing audit class rather than create a new one? Plus I thought we discussed refactoring the existing class and the methods inside it to generalise the audit method. It was not done properly before due to poorly specified requirements.

Reply:
The other class was getting quite big so I just stuck it in a new class for now - I could generalise the audit method so that we only have one class with a single method for everything

Reply:
I'll do that now

Reply:
there's now a single audit service now which you can feed and audit event into.